common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -109,7 +109,7 @@ static Level parseLevel(String levelStr) {
         Level level = Level.parseLevel(levelStr);
 
         MAX_RECORDS = SystemPropertyUtil.getInt(PROP_MAX_RECORDS, DEFAULT_MAX_RECORDS);
-
+        MAX_RECORDS_SAMPLED = MAX_RECORDS * 100;
```

Comment:
check for overflow? folks may set `MAX_RECORDS` to very high to make sure all records are captured.

Reply:
MAX_RECORDS_SAMPLED is a long but I think it may make sense to just cap it on `Integer.MAX_VALUE`